js/app.js

```diff
@@ -1,3 +1,107 @@
+    new Product('dog-duck.jpg'), new Product('dragon.jpg'), new Product('pen.jpg'), new Product('pet-sweep.jpg'),
+    new Product('scissors.jpg'), new Product('shark.jpg'), new Product('sweep.png'), new Product('tauntaun.jpg'),
+    new Product('unicorn.jpg'), new Product('water-can.jpg'), new Product('wine-glass.jpg'),
+];
+
+
+// Product constructor
+function Product(fileName) { // accepts new product as a filename e.g. bag.jpg
+    this.name = fileName.substring(0, fileName.indexOf('.')); // name of product dervived from file name
+    this.filePath = `img/${fileName}`; // storage location of product image file
+    this.countShown = 0; // counts number of times the product has been displayed for voting
+    this.countVoted = 0; // counts number of times the product has been chosen by the user
+    prodObjs.push(this);
+}
+
+// function to return a random index for allProducts array
+function rPG() { // the notorious Random Product Generator (this isn't nearly as clever as I think it is)
+    let products = []; // array to hold selected products to be displayed to user for voting
+    // generate disaplyQty number of randomly selected unique product images to be displayed to user
+    while (products.length < displayQty) {
+        // returns a whole number 0 - index of last item in allProducts array
+        let index = Math.floor(Math.random() * prodObjs.length);
+        while (!products.includes(prodObjs[index])) {
+            products.push(prodObjs[index]);
+            prodObjs[index].countShown++;
+        }
+    }
+    // display each of the selected products and insert attribute values
+    for (let i = 0; i < products.length; i++) {
+        let img = document.getElementById(`productImage${i}`);
+        img.src = products[i].filePath;
+        img.alt = products[i].name;
+        img.title = products[i].name;
+    }
+}
+
+// function to display View Results button after voteCount === maxVotes, and add event listener for said button
+function displayButton() {
+    // display button viewResultsBtn
+    document.getElementById('viewResults').innerHTML = '<input type="button" id="viewResultsBtn" value="View Results"/>';
+    // event listener to invole renderResults when button is clicked
+    let viewResultsBtn = document.getElementById('viewResultsBtn');
+    viewResultsBtn.addEventListener('click', renderResults);
+}
+
+// Event listener for product voting buttons
+let buttons = [document.getElementById('voteButton0'), document.getElementById('voteButton1'), document.getElementById('voteButton2')];
+for (let i = 0; i < buttons.length; i++) {
+    buttons[i].addEventListener('click', voteCounter);
+}
+
+
+// Event handler to increment vote trackers & remove event listener after maxVotes is reached
+function voteCounter(event) {
+    // increment vote counters on a user slection
+    if (event){
+        let target = event.target.alt;
+        for (let i = 0; i < prodObjs.length; i++) {
```

Comment:
If your products array were global, you could iterate over that instead of every product, which would make your code a bit more efficient.